WorkflowAdminBundle/Controller/Api/StatusController.php

```diff
@@ -82,21 +82,64 @@ public function deleteAction($statusId)
     {
         $status = $this->get('open_orchestra_model.repository.status')->find($statusId);
 
-        $canDelete = $this->authorizationChecker->isGranted(ContributionActionInterface::DELETE, $status)
-            && !$this->usageFinder->hasUsage($status)
-            && !$status->isInitialState()
-            && !$status->isPublishedState()
-            && !$status->isTranslationState()
-            && !$status->isAutoPublishFromState()
-            && !$status->isAutoUnpublishToState();
-
-        if ($canDelete) {
-            throw new DeleteStatusNotGrantedHttpException();
-        }
+        $this->denyDeleteUnlessGranted($status);
         $this->get('event_dispatcher')->dispatch(StatusEvents::STATUS_DELETE, new StatusEvent($status));
         $this->get('object_manager')->remove($status);
         $this->get('object_manager')->flush();
 
         return array();
     }
+
+    /**
+     * @param Request $request
+     *
+     * @Config\Route("/delete-multiple", name="open_orchestra_api_status_delete_multiple")
+     * @Config\Method({"DELETE"})
+     *
+     * @return Response
+     */
+    public function deleteStatusesAction(Request $request)
+    {
+        $format = $request->get('_format', 'json');
+
+        $facade = $this->get('jms_serializer')->deserialize(
+            $request->getContent(),
+            $this->getParameter('open_orchestra_workflow_admin.facade.status_collection.class'),
+            $format
+        );
+        $statusRepository = $this->get('open_orchestra_model.repository.status');
+        $statuses = $this->get('open_orchestra_api.transformer_manager')->get('status_collection')->reverseTransform($facade);
+
+        $statusIds = array();
+        foreach ($statuses as $status) {
+            $this->denyDeleteUnlessGranted($status);
```

Comment:
Should be harmonized with : 
https://github.com/open-orchestra/open-orchestra-cms-bundle/blob/master/UserAdminBundle/Controller/Api/UserController.php#L200